### PR TITLE
fix bug preventing IPv6 clusters using an existing VPC

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4628

We were accidentally wiping the VPC config for ipv6 clusters when validating the config file.

### CODE REVIEW

File: integration/tests/existing_vpc/existing_vpc_test.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func createVPC(stackName string, ctl api.ClusterProvider) *api.ClusterVPC {
 	publicSubnets, privateSubnets, vpcID, securityGroup := createVPCStackAndGetOutputs(stackName, ctl)
 
 	By("creating the cluster config from the existing VPC")
-	newVPC := api.NewClusterVPC()
+	newVPC := api.NewClusterVPC(false)
```
Comment (Skarlso) [inline integration/tests/existing_vpc/existing_vpc_test.go:139]: sorry, I'm blind. When is this ever true?

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 2]
```diff
@@ -777,7 +777,7 @@ func NewClusterConfig() *ClusterConfig {
 		KubernetesNetworkConfig: &KubernetesNetworkConfig{
 			IPFamily: DefaultIPFamily,
 		},
-		VPC: NewClusterVPC(),
+		VPC: NewClusterVPC(false),
```
Comment (Skarlso) [inline pkg/apis/eksctl.io/v1alpha5/types.go:780]: This... shouldn't be false here, should it?

File: integration/tests/existing_vpc/existing_vpc_test.go (hunk 1 above)
Comment (aclevername) [reply integration/tests/existing_vpc/existing_vpc_test.go:139]: its mainly used for testing. But if you wanted to generate a VPC config for ipv6 its needs to be true

File: pkg/apis/eksctl.io/v1alpha5/types.go (hunk 2 above)
Comment (aclevername) [reply pkg/apis/eksctl.io/v1alpha5/types.go:780]: the default new cluster is ipv4, so the cluster should contain an ipv4 vpc

File: integration/tests/existing_vpc/existing_vpc_test.go (hunk 1 above)
Comment (Skarlso) [reply integration/tests/existing_vpc/existing_vpc_test.go:139]: But how or where do you set that?

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 3]
```diff
@@ -788,15 +788,20 @@ func NewClusterConfig() *ClusterConfig {
 }
 
 // NewClusterVPC creates new VPC config for a cluster
-func NewClusterVPC() *ClusterVPC {
+func NewClusterVPC(ipv6Enabled bool) *ClusterVPC {
```
Comment (Skarlso) [inline pkg/apis/eksctl.io/v1alpha5/types.go:791]: Where is this ever called with true?

File: integration/tests/existing_vpc/existing_vpc_test.go (hunk 1 above)
Comment (Skarlso) [reply integration/tests/existing_vpc/existing_vpc_test.go:139]: Ahh! Found it. :D

```go
		if clusterConfig.VPC == nil {
			clusterConfig.VPC = api.NewClusterVPC(ipv6Enabled)
		}
```

File: pkg/apis/eksctl.io/v1alpha5/types.go (hunk 3 above)
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/types.go:791]: Never mind. :D Found it. I told you I'm blind. :D

File: pkg/ctl/cmdutils/configfile.go [hunk 4]
```diff
@@ -579,7 +577,7 @@ func NewUtilsEnableEndpointAccessLoader(cmd *Cmd, privateAccess, publicAccess bo
 	}
 	l.validateWithConfigFile = func() error {
 		if l.ClusterConfig.VPC == nil {
-			l.ClusterConfig.VPC = api.NewClusterVPC()
+			l.ClusterConfig.VPC = api.NewClusterVPC(false)
```
Comment (Skarlso) [inline pkg/ctl/cmdutils/configfile.go:580]: This couldn't be called on an IPv6 cluster? Why not do the same check here too? 🤔

Comment (aclevername) [reply pkg/ctl/cmdutils/configfile.go:580]: huh? Could you expand :smile:

Comment (Skarlso) [reply pkg/ctl/cmdutils/configfile.go:580]: In the above this setting is dependent on the ipv6 flag, here it's just false. What's the difference?

Comment (aclevername) [reply pkg/ctl/cmdutils/configfile.go:580]: this func isn't called for `createCluster`, so its not related

**Review by Skarlso — APPROVED** [review]

**Review by nikimanoledaki — APPROVED** [review]

**Pull request merged**